profile.py

```diff
@@ -150,8 +151,22 @@ def right_penumbra(self, upper=0.9, lower=0.1):
     def center(self, level=0.5):
         return 0.5 * (self.x_at_y(level) + self.x_at_y(level, reverse=True))
 
-    def mirror(self):
+    def mirror(self, m=0):
+        """
+        Provides mirror image of a profile with given Y axis (y=m).
+        Domain of profile might be changed due to this operation.
+
+        Use mirror() method to get values of mirrored profile:
+        >>> print((LateralProfile([[-1, 1], [0, -1], [1, 0]]).mirror()).y)
+        [ 0 -1  1]
+
+        :param m: Y value for mirror image.
+        :return: returns mirrored object
+        """
+        self.x = 2*m - self.x
+        self.x = self.x[::-1]
         self.y = self.y[::-1]
+        return self
```

Comment:
This method is doing mirroring in-place, modifying original object. It doesn't make sense to return reference to `self` in that case. Check for example `sort` method of class `list`.

```
l = [0,2,1]
x = l.sort()
print(l)
print(x)
```